python/lsst/faro/utils/tex_table.py

```diff
@@ -0,0 +1,616 @@
+        A `treecorr.KKCorrelation` object containing the correlation function.
+    """
+
+    xy = treecorr.KKCorrelation(**treecorrKwargs)
+    catA = treecorr.Catalog(ra=ra, dec=dec, k=k1, ra_units=raUnits, dec_units=decUnits)
+    if k2 is None:
+        # Calculate the auto-correlation
+        xy.process(catA)
+    else:
+        catB = treecorr.Catalog(
+            ra=ra, dec=dec, k=k2, ra_units=raUnits, dec_units=decUnits
+        )
+        # Calculate the cross-correlation
+        xy.process(catA, catB)
+
+    return xy
+
+
+def corrSpin2(
+    ra,
+    dec,
+    g1a,
+    g2a,
+    g1b=None,
+    g2b=None,
+    raUnits="degrees",
+    decUnits="degrees",
+    **treecorrKwargs
+):
+    """Function to compute correlations between at most two shear-like fields.
+    This is used to compute Rho statistics, given the appropriate spin-2
+    (shear-like) fields.
+    Parameters
+    ----------
+    ra : `numpy.array`
+        The right ascension values of entries in the catalog.
+    dec : `numpy.array`
+        The declination values of entries in the catalog.
+    g1a : `numpy.array`
+        The first component of the primary shear-like field.
+    g2a : `numpy.array`
+        The second component of the primary shear-like field.
+    g1b : `numpy.array`, optional
+        The first component of the secondary shear-like field.
+        Autocorrelation of the primary field is computed if `None` (default).
+    g2b : `numpy.array`, optional
+        The second component of the secondary shear-like field.
+        Autocorrelation of the primary field is computed if `None` (default).
+    raUnits : `str`, optional
+        Unit of the right ascension values.
+        Valid options are "degrees", "arcmin", "arcsec", "hours" or "radians".
+    decUnits : `str`, optional
+        Unit of the declination values.
+        Valid options are "degrees", "arcmin", "arcsec", "hours" or "radians".
+    **treecorrKwargs
+        Keyword arguments to be passed to `treecorr.GGCorrelation`.
+    Returns
+    -------
+    xy : `treecorr.GGCorrelation`
+        A `treecorr.GGCorrelation` object containing the correlation function.
+    """
+    xy = treecorr.GGCorrelation(**treecorrKwargs)
+    catA = treecorr.Catalog(
+        ra=ra, dec=dec, g1=g1a, g2=g2a, ra_units=raUnits, dec_units=decUnits
+    )
+    if g1b is None or g2b is None:
+        # Calculate the auto-correlation
+        xy.process(catA)
+    else:
+        catB = treecorr.Catalog(
+            ra=ra, dec=dec, g1=g1b, g2=g2b, ra_units=raUnits, dec_units=decUnits
+        )
+        # Calculate the cross-correlation
+        xy.process(catA, catB)
+
+    return xy
+
+
+def calculateTEx(catalog, config, prependString):
+    """Compute ellipticity residual correlation metrics using parquet table as
+    input. New column names based on these yaml specifications:
+    https://github.com/lsst/obs_lsst/blob/073901ab35efe5253ca64aeb14a290692ae400bd/policy/imsim/Source.yaml#L208-L226.
```

Comment:
Should this function have parameters and returns sections in the docstring?

Reply:
Updated.